source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/organizations/common/form.tsx

```diff
@@ -0,0 +1,279 @@
+
+import {
+  query,
+  buildOptions,
+  withLoader,
+  attributesFor,
+  UserResource,
+  OrganizationResource
+} from '@data';
+
+
+interface IOwnProps {
+  organization: OrganizationResource;
+  owner: UserResource;
+  users: UserResource[];
+  onSubmit: (attributes: IState, onSuccess?: () => void) => void;
+  onCancel: () => void;
+}
+
+interface IState {
+  name?: string;
+  websiteUrl?: string;
+  owner?: UserResource;
+  ownerError?: string;
+  buildEngineUrl?: string;
+  buildEngineApiAccessToken?: string;
+  logoUrl?: string;
+  publicByDefault?: boolean;
+}
+
+type IProps =
+  & i18nProps
+  & IOwnProps;
+
+
+@withTemplateHelpers
+class OrganizationForm extends React.Component<IProps, IState> {
+  mut: Mut;
+  toggle: Toggle;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const { organization, owner } = props;
+
+    if (organization) {
+
+      const {
+        name, websiteUrl, logoUrl,
+        buildEngineUrl, buildEngineApiAccessToken,
+        publicByDefault
+      } = attributesFor(organization);
+
+      this.state = {
+        name,
+        websiteUrl,
+        buildEngineUrl,
+        buildEngineApiAccessToken,
+        logoUrl,
+        publicByDefault,
+        owner
+      };
+
+    } else {
+      this.state = {
+        name: '',
+        websiteUrl: '',
+        buildEngineUrl: '',
+        buildEngineApiAccessToken: '',
+        logoUrl: '',
+        publicByDefault: false,
+        owner: null,
+        ownerError: ''
+      };
+    }
+
+  }
+
+  isValidForm = () => {
+    const { owner } = this.state;
+    if (!owner) {
+      this.setState({ ownerError: 'Cannot be empty' });
+    } else {
+      this.setState({ ownerError: '' });
+    }
+    return owner;
+  }
+
+  submit = async (e) => {
+    e.preventDefault();
+
+    const { onSubmit } = this.props;
+
+    if (this.isValidForm()) {
+      onSubmit(this.state, this.cleanForm);
+    }
+  }
+
+  cleanForm = () => {
+    this.setState({
+      name: '',
+      websiteUrl: '',
+      buildEngineUrl: '',
+      buildEngineApiAccessToken: '',
+      logoUrl: ''
+    });
+  }
+
+  cancel = (e) => {
+    e.preventDefault();
+    const { onCancel } = this.props;
+    onCancel();
+  }
+
+  ownerSelection = user => e => {
+    this.setState({
+      owner: user
+    });
+  }
+
+  render() {
+    const { mut, toggle } = this;
+
+    const {
+      name, websiteUrl, buildEngineUrl,
+      buildEngineApiAccessToken,
+      logoUrl, publicByDefault,
+      owner, ownerError
+    } = this.state;
+
+    const { t, users, organization } = this.props;
+
+    const { name: OwnerName } = attributesFor(owner);
+
+    return (
+      <>
+        <h2>
+          {
+            t(organization ?
+              'admin.settings.organizations.edit' :
+              'admin.settings.organizations.add'
+            )
+          }
+        </h2>
+        <div className='flex w-60'>
+          <form data-test-form className='ui form flex-grow'>
```

Comment:
data-test-form has _likely_ been used elsewhere. be sure when testing that there is more specificity than just this.
if there is another data-test-form on the page, it will become a part of your test when you may not want it to be